gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java

```diff
@@ -168,6 +167,7 @@ public class GobblinServiceManager implements ApplicationLauncher, StandardMetri
   @Inject
   @Getter
   private Injector injector;
+  @Getter @Setter private volatile static Injector staticInjector;
```

Comment:
use Injector instead of GobblinServiceGuiceModule because use of latter creates new instances of even classes marked as Singular and also takes really long time because it processes all the bindings and create injector again for every call to `getClass`

Reply:
wow this is a great comment and would not have known the reason for this. can you put it as a comment in the code?

Reply:
ok.
though because the reader of the code would not know that we were once using GobblinServiceGuiceModule